src/app/info/WikiItems.tsx

```diff
@@ -94,13 +115,98 @@ const WikiItems: React.FC<WikiItemsProps> = ({ rows }) => {
         const indexToRemove: number = displayRows.findIndex(
             (sortedRow) => row.wiki_key === sortedRow.wiki_key
         );
-        setDisplayRows((DisplayRows) => {
-            const temp = DisplayRows.slice();
+        setDisplayRows((displayRows) => {
+            const temp = displayRows.slice();
             temp.splice(indexToRemove, 1);
             return temp;
         });
         return indexToRemove;
     };
+
+    const [isSwapMode, setIsSwapMode] = useState(false);
+
+    const swapRows = async (row1: DbWikiRow, upwards: boolean): Promise<void> => {
+        if (isSwapMode) {
+            return;
+        }
+        setIsSwapMode(true);
+
+        const rowIndex1 = displayRows.findIndex((row) => {
+            return row.wiki_key == row1.wiki_key;
+        });
+        const rowIndex2 = rowIndex1 + (upwards ? -1 : 1);
+
+        if (rowIndex2 < 0 || rowIndex2 >= displayRows.length) {
+            setIsSwapMode(false);
+            return;
+        }
+
+        const row2 = displayRows[rowIndex2];
+
+        const { error } = await supabase.rpc("swap_two_wiki_rows", {
+            key1: row1.wiki_key,
+            key2: row2.wiki_key,
+        });
+
+        const auditLog = {
+            action: `move a wiki item ${row1.row_order <= row2.row_order ? "down" : "up"}`,
+            wikiId: row1.wiki_key,
+            content: {
+                itemName: row1.title,
+                oldRow_order: row1.row_order,
+            },
+        } as const satisfies Partial<AuditLog>;
```

Comment:
This is stronger typing than I'd expect here, is this required?

Reply:
Not sure but this is used every time an audit log is made, elsewhere in the codebase.

Reply:
Cool, I think it's a bit more than we need, but let's stick to pattern